openapi_server/apis/users_api.py

```diff
@@ -256,7 +266,27 @@ async def update_favorite_foods(
     ),
 ) -> ListFavoriteFoods:
     """This can only be done by the logged in user."""
-    ...
+    try:
+        user: DbUser = DbUser.get(username)
+    except DbUser.DoesNotExist:
+        raise HTTPException(status_code=404)
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+    fav_foods: List[DbFavoriteFood] = []  # to be saved in the database
+    favorite_foods: List[FavoriteFood] = []  # to be returned in the response
+    for item in list_favorite_foods.favorite_foods:
+        db_food: DbFavoriteFood = DbFavoriteFood(id=item.id, name=item.name)
+        fav_foods.append(db_food)
+        food: FavoriteFood = FavoriteFood(id=item.id, name=item.name)
+        favorite_foods.append(food)
+    user.favorite_foods = fav_foods
+    try:
+        user.save()
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+    return ListFavoriteFoods(favoriteFoods=favorite_foods)
```

Comment:
isn't list_favorite_foods.favorite_foods already ListFavoriteFoods type? why do we need to create a new variable(favorite_foods) and return the ListFavoriteFoods?

Reply:
I was thinking that I should manually create the response body following the logic in the for loop to ensure that the order and contents of the list is as expected. Like you mentioned, it might not be necessary to go through these extra steps. I can change line 289 to simply `return list_favorite_foods`, which is the request body supplied if that makes logical sense in a data assurance standpoint.

Reply:
@moyara I went ahead and made the line 289 change plus cleaning up a bit.